File: haskell/002/euler.hs
main :: IO ()

nextFibLessThan limit fibs = if (fibs !! (length fibs - 1)) + (fibs !! (length fibs - 2)) > limit
                               then fibs
                               else nextFibLessThan limit (fibs ++ [(fibs !! (length fibs - 1)) + (fibs !! (length fibs - 2))])

fibsLessThan limit = nextFibLessThan limit [0, 1]

main = print(sum(filter(\x -> mod x 2 == 0)(fibsLessThan 4000000)))

File: nim/002/euler.nim
from sequtils import `filter`
from math import `sum`

var fibs = @[0, 1]

while fibs[fibs.len - 1] + fibs[fibs.len - 2] < 4000000:
  add(fibs, fibs[fibs.len - 1] + fibs[fibs.len - 2])

echo(sum(filter(fibs, proc(x: int): bool = x mod 2 == 0)))
